test/dynamo/test_subclasses.py

```diff
@@ -2,26 +2,101 @@
 import contextlib
 
 import torch
+import torch._C
 
 import torch._dynamo.test_case
 import torch._dynamo.testing
 import torch._functorch.config
 import torch.utils.checkpoint
+from torch.utils._pytree import tree_map_only
 
 
-class MockSubclass(torch.Tensor):
+class PassthroughLeftAddSubclass(torch.Tensor):
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if func == torch.add:
+            return args[0]
+
+        return super().__torch_function__(func, types, args, kwargs)
+
+
+class PassthroughRightAddSubclassLeft(PassthroughLeftAddSubclass):
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if func == torch.add:
+            return args[1]
+
+        return super().__torch_function__(func, types, args, kwargs)
+
+
+class PassthroughRightAddSubclass(torch.Tensor):
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if func == torch.add:
+            return args[1]
+
+        return super().__torch_function__(func, types, args, kwargs)
+
+
+class PassthroughMulSubclass(torch.Tensor):
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        if kwargs is None:
+            kwargs = {}
+
+        if func == torch.mul:
+            return args[0]
+
+        return super().__torch_function__(func, types, args, kwargs)
+
+
+class WrapperSubclass:
+    def __init__(self, tensor):
+        self.tensor = tensor
+
     @classmethod
     def __torch_function__(cls, func, types, args=(), kwargs=None):
         if kwargs is None:
             kwargs = {}
+
+        args = tree_map_only(WrapperSubclass, lambda x: x.tensor, args)
```

Comment:
LazyAwaitable in torchrec uses torch.fx.node.map_aggregate instead of tree_map_only, it will be good to test against that as well